objcgen/embedder.cs

```diff
@@ -0,0 +1,935 @@
+						return universe.LoadFile (filename);
+				}
+				throw ErrorHelper.CreateError (13, $"Can't find the assembly '{resolve_args.Name}', referenced by '{resolve_args.RequestingAssembly.FullName}'.");
+			};
+
+			foreach (var arg in args) {
+				if (!File.Exists (arg))
+					throw ErrorHelper.CreateError (11, $"The assembly {arg} does not exist.");
+
+				Assemblies.Add (universe.LoadFile (arg));
+				Console.WriteLine ($"\tParsed '{arg}'");
+			}
+
+			// if not specified then we use the first specified assembly name
+			if (LibraryName == null)
+				LibraryName = Path.GetFileNameWithoutExtension (args[0]);
+
+			Console.WriteLine ("Processing assemblies...");
+			var p = new ObjCProcessor ();
+			p.Process (Assemblies);
+
+			Console.WriteLine ("Generating binding code...");
+			var g = new ObjCGenerator () {
+				Processor = p,
+				GenerateLinkerExclusions = EnableLinker,
+			};
+			g.Generate ();
+			g.Write (OutputDirectory);
+
+			var exe = typeof (Driver).Assembly;
+			foreach (var res in exe.GetManifestResourceNames ()) {
+				if (res == "main.c") {
+					// no main is needed for dylib and don't re-write an existing main.c file - it's a template
+					if (CompilationTarget != CompilationTarget.StaticLibrary || File.Exists ("main.c"))
+						continue;
+				}
+				var path = Path.Combine (OutputDirectory, res);
+				Console.WriteLine ($"\tGenerated: {path}");
+				using (var sw = new StreamWriter (path))
+					exe.GetManifestResourceStream (res).CopyTo (sw.BaseStream);
+			}
+			return 0;
+		}
+
+		static string xcode_app;
+		public static string XcodeApp
+		{
+			get
+			{
+				if (string.IsNullOrEmpty (xcode_app)) {
+					int exitCode;
+					string output;
+					if (!Utils.RunProcess ("xcode-select", "-p", out exitCode, out output))
+						throw ErrorHelper.CreateError (6, "Could not find the Xcode location.");
+					xcode_app = Path.GetDirectoryName (Path.GetDirectoryName (output.Trim ()));
+				}
+				return xcode_app;
+			}
+		}
+
+		class BuildInfo
+		{
+			public string Sdk;
+			public string[] Architectures;
+			public string SdkName; // used in -m{SdkName}-version-min
+			public string MinVersion;
+			public string XamariniOSSDK;
+			public string CompilerFlags;
+			public string LinkerFlags;
+
+			public bool IsSimulator
+			{
+				get { return Sdk.Contains ("Simulator"); }
+			}
+		}
+
+		void VerifyDependencies ()
+		{
+			SystemCheck.VerifyMono ();
+			switch (Platform) {
+				case Platform.iOS:
+				case Platform.tvOS:
+				case Platform.watchOS:
+					SystemCheck.VerifyXamariniOS ();
+					break;
+				case Platform.macOS:
+				case Platform.macOSFull:
+				case Platform.macOSModern:
+				case Platform.macOSSystem:
+					SystemCheck.VerifyXamarinMac ();
+					break;
+				default:
+					throw ErrorHelper.CreateError (99, "Internal error: invalid platform {0}. Please file a bug report with a test case (https://github.com/mono/Embeddinator-4000/issues).", Platform);
+			}
+		}
+
+		public int Compile ()
```

Comment:
I think we should refactor the class into smaller bits instead of splitting partial file, but we can do that later (since there won't be too much traffic in embedder.cs in main hopefully).